packages/contractkit/src/sortedlinkedlist.ts

```diff
@@ -0,0 +1,21 @@
+import { BigNumber } from 'bignumber.js'
+
+export const NULL_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+export interface ListElement {
+  key: string
+  value: BigNumber
+}
+
+/**
+ * Returns the two keys around element
+ * @param elem Element to find lesser and greater keys for
+ * @param sorted Descending sorted list
+ */
+export function getLesserAndGreaterKeys(elem: ListElement, sorted: ListElement[]) {
+  const lesserElem = sorted.find((e) => e.value.lt(elem.value) && e.key !== elem.key)
+  const lesserKey = lesserElem === undefined ? NULL_ADDRESS : lesserElem.key
+  const greaterElem = sorted.reverse().find((e) => e.value.gt(elem.value) && e.key !== elem.key)
```

Comment:
How long do we expect these sorted lists to get? If they're likely to be very long, and we're running this function frequently, it could be worth avoiding the extra step of reversing the list.

Also, calling `reverse()` on an array modifies it in place. Is this list used anywhere else that assumes it to still be sorted in descending order? If so, `reverse()` should be avoided here.

You can find both `lesserElem` and `greaterElem` in a single pass through the list in a way that avoids modifying the input: 

```typescript
let lesserElem = null
let greaterElem = null

for (const currentElement of sorted) {
  if (currentElement.value.gt(elem.value)) {
    greaterElem = currentElement // keep resetting greater element until we get to the element
  } else if (currentElement.value.lt(elem.value)) {
    lesserElem = currentElement
    break // since it's a descending list, we can stop as soon as we find the first thing less than the element
  }
}
```

Note: you can skip `e.key !== elem.key` because `elem.value.gt(elem.value)` or `elem.value.lt(elem.value)` will never be true. You'd only need to do this check if you were using `gte()` or `lte()`.